flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/GroupAggProcessFunction.scala

```diff
@@ -118,19 +131,33 @@ class GroupAggProcessFunction(
       }
     }
 
-    state.update(accumulators)
+    if (inputCnt > 0) {
```

Comment:
hi, we need to fix the deduction of `generateRetraction` also. Set `generateRetraction` to true only when `DataStreamGroupAggregate ` is under `needRetraction` mode. 
Remove `override def consumesRetractions = true` so that we can get the correct AccMode. A `DataStreamGroupAggregate ` may in AccRetract mode even if it's `generateRetraction ` is false(last retract). The correct AccMode will provides import information to let downstream operators choose correct acc or retraction functions.